addons/mail/wizard/mail_compose_message.py

```diff
@@ -383,6 +385,16 @@ def _compute_auto_delete_keep_log(self):
         toreset.auto_delete_keep_log = False
         (self - toreset).auto_delete_keep_log = True
 
+    @api.depends('template_id')
+    def _compute_mail_server_id(self):
+        """ Copy value from template when updating it, if set on template. When
+        removing the template, reset it. """
+        for composer in self:
+            if composer.template_id.mail_server_id:
+                composer.mail_server_id = composer.template_id.mail_server_id
+            if not composer.template_id:
+                composer.mail_server_id = False
```

Comment:
Is this part necessary? 

`composer.mail_server_id = composer.template_id.mail_server_id` should be all you need as an empty recordset will return a Falsy value anyway?